llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -1213,6 +1214,27 @@ static SDValue performSHLCombine(SDNode *N, SelectionDAG &DAG,
                      DAG.getConstant(SMSize, DL, MVT::i32));
 }
 
+static SDValue performSignExtendCombine(SDNode *N, SelectionDAG &DAG,
+                                        TargetLowering::DAGCombinerInfo &DCI,
+                                        const MipsSubtarget &Subtarget) {
+  SDValue N0 = N->getOperand(0);
+  EVT VT = N->getValueType(0);
+
+  // Pattern match XOR.
+  //  $dst = sign_extend (xor (trunc $src), imm)
```

Comment:
`trunc` needs an argument.  I guess the argument will have some effects on the result.

Reply:
For example, if `$src=0x12345678`; `imm=0xffff`.

lets `truncate it to i8`, then it will be `0x78`,  the result will be ~0x78.
lets `truncate it to i16`, then it will be `0x5678`,  the result will be ~0x5678.